src/harvester/doris_bfs.rs

```diff
@@ -119,52 +120,42 @@ async fn fetch_dataset(dir: &Dir, client: &Client, source: &Source, handle: &str
     })
     .await?;
 
-    let mut urn = String::new();
-    let mut title = String::new();
-    let mut description = String::new();
+    let identifier;
+    let title;
+    let r#abstract;
 
     {
         let document = Html::parse_document(&response);
 
-        for element in document.select(&SELECTORS.row_selector) {
-            let label_element = match element.select(&SELECTORS.label_selector).next() {
-                Some(label_element) => label_element,
-                None => continue,
-            };
-
-            let label = label_element.text().collect::<String>();
-
-            let value_element = match element.select(&SELECTORS.value_selector).next() {
-                Some(value_element) => value_element,
-                None => continue,
-            };
-
-            let value = value_element.text().collect::<String>();
-
-            match label.trim() {
-                "URN(s):" => urn = value.trim().to_owned(),
-                "Titel:" => title = value.trim().to_owned(),
-                "Zusammenfassung:" => description = value.trim().to_owned(),
-                _ => (),
-            }
-        }
+        identifier = document
+            .select(&SELECTORS.identifier_selector)
+            .filter_map(|element| element.value().attr("content"))
+            .find(|identifier| identifier.starts_with("urn:"))
+            .ok_or_else(|| anyhow!("Missing identifier"))?
+            .to_owned();
+
+        title = document
+            .select(&SELECTORS.title_selector)
+            .next()
+            .and_then(|element| element.value().attr("content"))
+            .ok_or_else(|| anyhow!("Missing title"))?
+            .to_owned();
+
+        r#abstract = document
+            .select(&SELECTORS.abstract_selector)
+            .next()
+            .and_then(|element| element.value().attr("content"))
+            .unwrap_or_default()
```

Comment:
Why is ther no ok_or_else here?

Reply:
Because I considered the abstract optional in our internal schema.